### ISSUE TITLE
jcv_fillgaps crashes on specific point configuration

### ISSUE THREAD

**Issue by kylemcdonald** [JCash/voronoi#7]
```
Assertion failed: (internal->numsites == 1), function jcv_fillgaps, file jc_voronoi.h, line 1125.
fish: './relax -r 10 <test.csv >output…' terminated by signal SIGABRT (Abort)
```

Example file attached: [test.csv.zip](https://github.com/JCash/voronoi/files/884170/test.csv.zip)

Here is a visualization of where the two problematic points are:

<img width="714" alt="z0btmaaaaasuvork5cyii -1" src="https://cloud.githubusercontent.com/assets/157106/24531833/b061f7c8-1570-11e7-8275-9f7dca811192.png">

If I remove either of these points, the problem goes away.

**Comment by JCash** [comment]
Thanks for the report @kylemcdonald and for the test data!

The problem was that points outside of the bounding box (0,0),(width,height) were never really supported properly, and that was a poor design choice by me.

I've updated the api to take a bounding rectangle (or null) and the function now prunes the input points based on the bounding box.

The documentation was updated with a note about this.

Feel free to test these fixes on the https://github.com/JCash/voronoi/tree/dev branch
I'll probably merge into master in a couple of days.
